service/controller/resource/tccpf/create.go

```diff
@@ -170,10 +170,11 @@ func (r *Resource) newRecordSetsParams(ctx context.Context, cr infrastructurev1a
 	var recordSets *template.ParamsMainRecordSets
 	{
 		recordSets = &template.ParamsMainRecordSets{
-			BaseDomain:                 key.ClusterBaseDomain(cr),
-			ClusterID:                  key.ClusterID(&cr),
-			GuestHostedZoneNameServers: cc.Status.TenantCluster.DNS.HostedZoneNameServers,
-			Route53Enabled:             r.route53Enabled,
+			BaseDomain:                     key.ClusterBaseDomain(cr),
+			ClusterID:                      key.ClusterID(&cr),
+			ControlPlanePublicHostedZoneID: cc.Status.ControlPlane.HostedZone.ID,
```

Comment:
When this is used here, should we have a check at the beginning of `EnsureCreated` to cancel if the ID is not yet present.

Reply:
I just wrote the check and then realised it should be always available (this is CP HostedZone ID) unless Route53 is disabled (in which case we need to ignore empty value anyway).